// @aidprotocol/trust-compute/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  DEFAULT_WEIGHTS: () => DEFAULT_WEIGHTS,
  FORMULA_VERSION: () => FORMULA_VERSION,
  HASH_ALGORITHM: () => HASH_ALGORITHM,
  computeTrustScore: () => computeTrustScore,
  getTrustVerdict: () => getTrustVerdict,
  jcsSerialize: () => jcsSerialize,
  verifyTrustProof: () => verifyTrustProof
});
module.exports = __toCommonJS(src_exports);
var import_crypto = require("crypto");
var HASH_ALGORITHM = "sha384";
var FORMULA_VERSION = "1.0.0";
var DEFAULT_WEIGHTS = {
  successRate: 40,
  chainCoverage: 25,
  volume: 20,
  manifestAdherence: 15
};
function jcsSerialize(value) {
  if (value === null)
    return "null";
  if (typeof value === "boolean")
    return value ? "true" : "false";
  if (typeof value === "number") {
    if (!isFinite(value))
      throw new Error("JCS: non-finite numbers not supported");
    return Object.is(value, -0) ? "0" : String(value);
  }
  if (typeof value === "string")
    return JSON.stringify(value);
  if (Array.isArray(value)) {
    return "[" + value.map(jcsSerialize).join(",") + "]";
  }
  if (typeof value === "object") {
    const obj = value;
    const keys = Object.keys(obj).filter((k) => obj[k] !== void 0).sort();
    const entries = keys.map((k) => JSON.stringify(k) + ":" + jcsSerialize(obj[k]));
    return "{" + entries.join(",") + "}";
  }
  return "";
}
function computeTrustScore(stats, weights = DEFAULT_WEIGHTS) {
  const volumeScore = Math.min(stats.attestationCount / 1e3, 1);
  const manifestScore = stats.manifestAdherence > 0 || stats.attestationCount > 0 ? stats.manifestAdherence : 0.5;
  const score = Math.round(
    stats.successRate * weights.successRate + stats.chainCoverage * weights.chainCoverage + Math.min(volumeScore, 1) * weights.volume + manifestScore * weights.manifestAdherence
  );
  const clampedScore = Math.max(0, Math.min(100, score));
  const proofData = { inputs: stats, weights, score: clampedScore };
  const canonical = jcsSerialize(proofData);
  const proofHash = (0, import_crypto.createHash)(HASH_ALGORITHM).update(canonical).digest("hex");
  return {
    score: clampedScore,
    inputs: stats,
    weights,
    proofHash,
    formulaVersion: FORMULA_VERSION,
    hashAlgorithm: HASH_ALGORITHM
  };
}
function getTrustVerdict(score) {
  if (score >= 90)
    return { verdict: "proceed", discount: 0.3, settlementMode: "deferred" };
  if (score >= 80)
    return { verdict: "trusted", discount: 0.25, settlementMode: "batched" };
  if (score >= 60)
    return { verdict: "standard", discount: 0.2, settlementMode: "batched" };
  if (score >= 40)
    return { verdict: "caution", discount: 0.1, settlementMode: "standard" };
  if (score >= 20)
    return { verdict: "building", discount: 0, settlementMode: "immediate" };
  return { verdict: "new", discount: 0, settlementMode: "immediate" };
}
function verifyTrustProof(proof) {
  const recomputed = computeTrustScore(proof.inputs, proof.weights);
  return recomputed.proofHash === proof.proofHash && recomputed.score === proof.score;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DEFAULT_WEIGHTS,
  FORMULA_VERSION,
  HASH_ALGORITHM,
  computeTrustScore,
  getTrustVerdict,
  jcsSerialize,
  verifyTrustProof
});
/**
 * @aidprotocol/trust-compute — Standalone AID trust scoring library
 *
 * Deterministic trust score computation for AI agents.
 * No dependencies. No database. Pure computation.
 *
 * Given attestation stats, produces a trust score + cryptographic proof hash.
 * Anyone can run this to independently verify scores published by any AID oracle.
 *
 * Uses SHA-384 for proof hashes (quantum-resistant, NIST PQC migration ready).
 * Algorithm-agile: hashAlgorithm field in output enables future migration.
 *
 * @license MIT
 * @see https://claw-net.org
 */
