// @aidprotocol/trust-compute/dist/index.mjs
// src/index.ts
import { createHash } from "crypto";
var HASH_ALGORITHM = "sha384";
var FORMULA_VERSION = "1.0.0";
var DEFAULT_WEIGHTS = {
  successRate: 40,
  chainCoverage: 25,
  volume: 20,
  manifestAdherence: 15
};
function jcsSerialize(value) {
  if (value === null)
    return "null";
  if (typeof value === "boolean")
    return value ? "true" : "false";
  if (typeof value === "number") {
    if (!isFinite(value))
      throw new Error("JCS: non-finite numbers not supported");
    return Object.is(value, -0) ? "0" : String(value);
  }
  if (typeof value === "string")
    return JSON.stringify(value);
  if (Array.isArray(value)) {
    return "[" + value.map(jcsSerialize).join(",") + "]";
  }
  if (typeof value === "object") {
    const obj = value;
    const keys = Object.keys(obj).filter((k) => obj[k] !== void 0).sort();
    const entries = keys.map((k) => JSON.stringify(k) + ":" + jcsSerialize(obj[k]));
    return "{" + entries.join(",") + "}";
  }
  return "";
}
function computeTrustScore(stats, weights = DEFAULT_WEIGHTS) {
  const volumeScore = Math.min(stats.attestationCount / 1e3, 1);
  const manifestScore = stats.manifestAdherence > 0 || stats.attestationCount > 0 ? stats.manifestAdherence : 0.5;
  const score = Math.round(
    stats.successRate * weights.successRate + stats.chainCoverage * weights.chainCoverage + Math.min(volumeScore, 1) * weights.volume + manifestScore * weights.manifestAdherence
  );
  const clampedScore = Math.max(0, Math.min(100, score));
  const proofData = { inputs: stats, weights, score: clampedScore };
  const canonical = jcsSerialize(proofData);
  const proofHash = createHash(HASH_ALGORITHM).update(canonical).digest("hex");
  return {
    score: clampedScore,
    inputs: stats,
    weights,
    proofHash,
    formulaVersion: FORMULA_VERSION,
    hashAlgorithm: HASH_ALGORITHM
  };
}
function getTrustVerdict(score) {
  if (score >= 90)
    return { verdict: "proceed", discount: 0.3, settlementMode: "deferred" };
  if (score >= 80)
    return { verdict: "trusted", discount: 0.25, settlementMode: "batched" };
  if (score >= 60)
    return { verdict: "standard", discount: 0.2, settlementMode: "batched" };
  if (score >= 40)
    return { verdict: "caution", discount: 0.1, settlementMode: "standard" };
  if (score >= 20)
    return { verdict: "building", discount: 0, settlementMode: "immediate" };
  return { verdict: "new", discount: 0, settlementMode: "immediate" };
}
function verifyTrustProof(proof) {
  const recomputed = computeTrustScore(proof.inputs, proof.weights);
  return recomputed.proofHash === proof.proofHash && recomputed.score === proof.score;
}
export {
  DEFAULT_WEIGHTS,
  FORMULA_VERSION,
  HASH_ALGORITHM,
  computeTrustScore,
  getTrustVerdict,
  jcsSerialize,
  verifyTrustProof
};
/**
 * @aidprotocol/trust-compute — Standalone AID trust scoring library
 *
 * Deterministic trust score computation for AI agents.
 * No dependencies. No database. Pure computation.
 *
 * Given attestation stats, produces a trust score + cryptographic proof hash.
 * Anyone can run this to independently verify scores published by any AID oracle.
 *
 * Uses SHA-384 for proof hashes (quantum-resistant, NIST PQC migration ready).
 * Algorithm-agile: hashAlgorithm field in output enables future migration.
 *
 * @license MIT
 * @see https://claw-net.org
 */
